pathway-analysis/01-overrepresentation_analysis.Rmd

```diff
@@ -230,28 +263,54 @@ fisher.test(gene_table, alternative = "greater")
+Our DGE results are from data published as part of [Sachs _et al._ (2014)](https://dx.doi.org/10.1182%2Fblood-2013-08-521708).
+The authors sorted populations of primary leukemia cells and examined the stem cell capacity of these cell populations.
+(This study may sound familiar if you've worked on one of our bulk RNA-seq exercise notebooks in the past!)
 
-```{r top_neuron_genes, live = TRUE}
-# Select genes that are in the top 100 for at least one comparison to other
-# cell types
-top_neuron_genes <- neuron_markers_df %>%
-  dplyr::filter(Top <= 100) %>%
-  dplyr::pull(gene_symbol)
+We compared the population that the authors identified as having high stem cell capacity to a low stem cell capacity population.
+We also compared the high stem cell capacity cells to a mix of populations (e.g., unsorted cells).
+You can see the code in [here](https://github.com/AlexsLemonade/training-modules/tree/master/pathway-analysis/setup/03-leukemia_DGE.Rmd).
+<!--TODO: this is about to get renumbered-->
+
+We're interested in what pathways are over-represented in genes that specifically distinguish the high capacity population from the low capacity population.
+
+Let's generate a list of genes that have higher expression in the high stem cell capacity population compared to the low stem cell capacity population, but we'll also want to exclude genes that show up in our _other comparison_ to unsorted cells.
 
-# Because genes can theoretically be listed as many times as there are other
-# cell types, we need to filter to unique genes
-top_neuron_genes <- unique(as.character(top_neuron_genes))
+We'll start with the high stem cell capacity vs. low stem cell capacity population comparison.
+Genes with positive log2 fold-changes (LFC) will be more highly expressed in the high stem cell capacity cells based on how we set up the analysis.
+
+```{r high_capacity_genes}
+vs_low_genes <- vs_low_df %>%
+  # Filter to the positive LFC and filter based on significance too (padj)
+  dplyr::filter(log2FoldChange > 0,
+                padj < 0.05) %>%
+  # Return a vector of gene symbols
+  dplyr::pull(gene_symbol)
 ```
 
-Maybe you're thinking, "Hey, picking the top 100 genes for each comparison seems kind of arbitrary."
-You are correct!
+Although we're picking a _commonly used_ cutoff (FDR < 0.05), it's still arbitrary and we could just as easily pick a different threshold for our LFC values.
 When we generate lists of genes of interest for ORA, we typically pick an arbitrary cutoff.
 This is one of the approach's weaknesses -- we've removed all other context.
 
+Now, we'll take the same steps for our other results. 
+
+```{r unsorted_genes_to_remove, live = TRUE}
+vs_unsorted_genes <- vs_unsorted_df %>%
+  dplyr::filter(log2FoldChange > 0,
+                padj < 0.05) %>%
+  dplyr::pull(gene_symbol)
+```
+
+We want genes that are in the first comparison but not in the second!
+We can use `setdiff()`, a base R function for set operations, to get the list that we want.
```

Comment:
Probably not something people will think much about, but I just had to look to satisfy myself that order mattered for `setdiff()`

Reply:
The comment in the code block itself is a reminder for me to say that during training. Observation: I believe you are of the opinion that people are more likely to read the text outside the comments than the comments themslves and I have the opposite opinion. My opinion is founded pretty much entirely in my own behavior/tendencies.